### PR TITLE
Stellar address validation 

### PR DESCRIPTION
**Pull request by 0oM4R**
### Description

we have two places that we need to validate stellar addresses, `Your farms` and `Withdraw` dialog

### Changes

- moved the addressValidation logic to validators
- Used form validator and Input validator component to use async rules 
- applied the address validation rules on both your farms and withdraw dialogs

### Related Issues

- #2286 
- #2343

### Documentation PR

For UI changes, Plaese provide the Documetation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/utils/validators.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
+import { default as StellarSdk, StrKey } from "stellar-sdk";
```
Comment (MohamedElmdary) [inline packages/playground/src/utils/validators.ts:1]:
```suggestion
import StellarSdk from "stellar-sdk";
```

we don't use `StrKey` here

File: packages/playground/src/dashboard/components/user_farms.vue [hunk 2]
```diff
@@ -99,12 +99,14 @@
                   <input-validator
                     :value="address"
                     :rules="[validators.required('Address is required.'), customStellarValidation]"
+                    :async-rules="[() => validators.isValidStellarAddress(address)]"
```
Comment (MohamedElmdary) [inline packages/playground/src/dashboard/components/user_farms.vue:102]:
```suggestion
                    :async-rules="[validators.isValidStellarAddress]"
```

we should be able to pass a ref to function directly

File: packages/playground/src/components/withdraw_dialog.vue [hunk 3]
```diff
@@ -13,33 +13,49 @@
...
             >
-            </v-text-field>
-          </v-form>
+              <v-text-field
+                v-bind="{ ...validationProps }"
+                @paste.prevent
+                label="Amount (TFT)"
+                v-model="amount"
+                type="number"
+                onkeydown="javascript: return event.keyCode == 69 || /^\+$/.test(event.key) ? false : true"
```
Comment (MohamedElmdary) [inline packages/playground/src/components/withdraw_dialog.vue:54]: what is this code used for .... I'm pretty sure we can remove it and code should last working as before

Comment (0oM4R) [reply packages/playground/src/components/withdraw_dialog.vue:54]: after discussion with @MohamedElmdary we agreed that this line is functional and it prevents user from typing `e` and +

**Review by MohamedElmdary — APPROVED** [review]

File: packages/playground/src/components/withdraw_dialog.vue [hunk 4]
```diff
@@ -94,53 +109,28 @@ onMounted(async () => {
...
+    return {
+      message: "Blocked Address",
+    };
   }
   if (!isValid || target.value.match(/\W/)) {
-    targetError.value = "invalid address";
-    isValidSwap.value = false;
-    return false;
+    return {
+      message: "invalid address",
```
Comment (zaelgohary) [inline packages/playground/src/components/withdraw_dialog.vue:126]:
```suggestion
      message: "Invalid address",
```

File: packages/playground/src/components/withdraw_dialog.vue [hunk 5]
```diff
@@ -13,33 +13,49 @@
...
-                () => amount < freeBalance! || 'Amount cannot exceed balance',
-              ]"
+              validators.required('This field is required'),
+              validators.min('Amount should be at least 2 TFT', 2),
+              validators.max( 'Amount cannot exceed balance',freeBalance!),
+              () => {
+                if(!(amount.toString().split('.').length > 1 ? amount.toString().split('.')[1].length <= 3 : true)){
+                  return {
+                    message:  'Amount must have 3 decimals only'
+                  }
```
Comment (zaelgohary) [inline packages/playground/src/components/withdraw_dialog.vue:43]: Can we remove this to a separate function in the script or in utils?

File: packages/playground/src/utils/validators.ts [hunk 6]
```diff
@@ -739,3 +742,22 @@ export function isValidSSHKey(key: string): boolean {
...
+  const server = new StellarSdk.Server(window.env.STELLAR_HORIZON_URL);
+  try {
+    // check if the account provided exists on stellar
+    const account = await server.loadAccount(target);
+    // check if the account provided has the appropriate trustlines
+    const includes = account.balances.find(
+      (b: { asset_code: string; asset_issuer: string }) =>
+        b.asset_code === "TFT" && b.asset_issuer === window.env.TFT_ASSET_ISSUER,
+    );
+    if (!includes) throw new Error("invalid trustline");
```
Comment (zaelgohary) [inline packages/playground/src/utils/validators.ts:755]:
```suggestion
    if (!includes) throw new Error("Invalid trustline");
```

File: packages/playground/src/utils/validators.ts [hunk 7]
```diff
@@ -739,3 +742,22 @@ export function isValidSSHKey(key: string): boolean {
...
+  } catch (e) {
+    const message =
+      (e as Error).message === "invalid trustline"
```
Comment (zaelgohary) [inline packages/playground/src/utils/validators.ts:758]:
```suggestion
      (e as Error).message === "Invalid trustline"
```

**Review by zaelgohary — COMMENTED** [review]
Good job, Omar. just some small comments.

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**